internal/commands/buildpack_new.go

```diff
@@ -85,6 +99,14 @@ func BuildpackNew(logger logging.Logger, creator BuildpackCreator) *cobra.Comman
 	cmd.Flags().StringVarP(&flags.Path, "path", "p", "", "Path to generate the buildpack")
 	cmd.Flags().StringVarP(&flags.Version, "version", "V", "1.0.0", "Version of the generated buildpack")
 	cmd.Flags().StringSliceVarP(&flags.Stacks, "stacks", "s", []string{"io.buildpacks.stacks.jammy"}, "Stack(s) this buildpack will be compatible with"+stringSliceHelp("stack"))
+	cmd.Flags().MarkDeprecated("stacks", "stacks is deprecated in the favor of `targets`")
+	cmd.Flags().Var(&flags.Targets, "targets",
```

Comment:
What we've done in other parts of the code is to define a flag that receives `[]string` and then we parse each individual string to convert it into a struct.

For example, if you add

```go
type BuildpackNewFlags struct {
    //  ... other fields	
    Targets  []string
}
```

Then you can have a method to parse the targets, in this method, you can `strings.split` by `:` to divide the target on:
1.  `os/arch/variant`
   -  Here you can `strings.split` by `/`
2. `name@version` 
  - Here you can `strings.split` by ';' (based on the new format suggested by you)
      - `string.split` by '@' 
After having all the pieces you can instantiate your `Target` structure.

I think this can simplify all the logic you have to create the `Target`